src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -321,138 +384,104 @@ int MPL_gavl_tree_free(MPL_gavl_tree_t gavl_tree)
                 else
                     cur_node->right = NULL;
             }
-            if (gavl_tree_iptr->gavl_free_fn)
-                gavl_tree_iptr->gavl_free_fn((void *) dnode->val);
+            if (tree_ptr->gavl_free_fn)
+                tree_ptr->gavl_free_fn((void *) dnode->val);
             MPL_free(dnode);
         }
     }
-    MPL_free(gavl_tree_iptr);
+    MPL_free(tree_ptr);
     return mpl_err;
 }
 
-int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t len)
+static void gavl_tree_delete_node(gavl_tree_s * tree_ptr, gavl_tree_node_s * dnode, void **val)
 {
-    int mpl_err = MPL_SUCCESS;
-    gavl_tree_node_s *cur_node;
     gavl_tree_node_s *inorder_node;
-    gavl_tree_s *gavl_tree_iptr = (gavl_tree_s *) gavl_tree;
 
-    DECLARE_STACK(gavl_tree_node_s *, node_stack);
+    if (dnode->right == NULL) {
+        /* no right child, next inorder node is parent */
+        if (dnode->parent == NULL) {
+            /* delete root node */
+            if (dnode->left) {
```

Comment:
Can you make `delete root node` more specific? E.g.,delete root node and assign the left child as the new root.